packages/react-instantsearch-nextjs/src/InstantSearchNext.tsx

```diff
@@ -72,7 +73,9 @@ This message will only be displayed in development mode.`
     <InstantSearchRSCContext.Provider value={promiseRef}>
       <InstantSearchSSRProvider initialResults={initialResults}>
         <InstantSearch {...instantSearchProps} routing={routing}>
-          {!initialResults && <InitializePromise />}
+          {!initialResults && (
+            <InitializePromise nonce={headers().get('x-nonce') || undefined} />
```

Comment:
`nonce` could be `string | null` to avoid an explicit `undefined`

Reply:
It still needs to be `string | undefined` in the injected script. I was ambivalent between making the fallback here or in InitializePromise.